src/cpp/channel.cpp

```diff
@@ -43,9 +43,10 @@ bool Channel::push(const sol::variadic_args& args) {
         }
         RETHROW_WITH_PREFIX("effil.channel:push");
     }
-    if (ctx_->channel_.empty())
-        ctx_->cv_.notify_one();
+    const bool wasEmpty = ctx_->channel_.empty();
```

Comment:
What is the problem here?

Reply:
Race condition. Channel is empty -> we trigger a condition variable -> another thread is fired -> another thread sees that channel is empty and starts waiting again -> we finally put data to channel but do not notify again -> another thread awaits to be triggered forever